core/templates/pages/donate-page/donate-page.component.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the donate page.
+ */
+
+require('base-components/base-content.directive.ts');
+
+require('domain/utilities/url-interpolation.service.ts');
+require('services/site-analytics.service.ts');
+require('services/contextual/window-dimensions.service.ts');
+
+angular.module('oppia').component('donatePage', {
+  bindings: {},
+  template: require('./donate-page.component.html'),
```

Comment:
Hi @vojtechjelinek, 

I've changed the filename from `donate-page.directive.html` to `donate-page.component.html` and it gives me this error `[$injector:unpr] http://errors.angularjs.org/1.7.9/$injector/unpr?p0=objProvider%20%3C-%20obj` on the donate page. I think this has something to do with how webpack bundles are built. Do you have any suggestion to fix this?

Reply:
In order to use `require` you will need to add the `component.html` to https://github.com/oppia/oppia/blob/develop/webpack.common.config.ts#L691

Reply:
@vojtechjelinek something on the lines of the changes in this pr? https://github.com/oppia/oppia/pull/9156/files (the webpack.common.config.ts file). The very last file listed.

Reply:
Yep, something like that, it could also be a part of the same Regex as @nishantwrp shown it the email thread.